### ISSUE TITLE
OnIowa event locations

### ISSUE THREAD

**Issue by briand44** [uiowa/uiowa#6402]
# Story

As an OnIowa website owner, I would like to be able to reuse locations for events happening during the same year as well as from year to year so that we do not have to enter the same location information each time.

# Additional Information

They would like the ability to have a location title/name for places like Hubbard Park
They would like the ability to store lat/long coordinates for a location.
They would like the ability to provide additional direction information for a location.
The location information will need to be made available in the API consumed by the Hawktools app in a future slice.

# Proposed Solution

- Add replicate functionality to SiteNow events (local drupal events)
- Update company name label to location name
- Add lat/long geofields
- Add additional directions field
- Create an internally focused display that shows existing events on their site with the location where they can easily find and get to an appropriate event to replicate

**Comment by bspeare** [comment]
Content hub seems to have this feature.  Is it worth exploring having them use private events from content.uiowa.edu?

**Comment by briand44** [comment]
We suggested that initially but they don't want to use content.uiowa.edu

**Comment by joewhitsitt** [comment]
I feel like we should override the "company" field of address, add a lat/long (geofield) and add a directions field. I thought there was a separate issue to investigate the JSON or XML output

I think their could be some kind of autofill functionality of address information to supplement but not be in place of.

**Comment by GaryRidgway** [comment]
Potentially an RSS feed from facilities to ensure that we have canonical links back to the original building if used as an autocomplete. That would allow the content to be managed by those that already manage it and to keep erroneous entries light.

**Comment by briand44** [comment]
I can reach out to OnIowa and see if the additions to the existing events (lat/long, additional directions, etc) and the ability to replicate events will meet their needs.

**Comment by briand44** [comment]
>I thought there was a separate issue to investigate the JSON or XML output

@joewhitsitt there is. I was just mentioning it here to make sure we had that in mind when we decide on how to move forward with this. I updated that statement to make it more clear that is in a future slice.

**Comment by briand44** [comment]
I've reached out for feedback on the proposed solution.

**Comment by joewhitsitt** [comment]
My original from last grooming of using content hub was only solve the saved locations that would autofill (not store it on oniowa), not rip out the local fields for capturing location info. So they could still use a saved location but customize it to say “east side steps”

**Comment by briand44** [comment]
Grooming conversation... Let's probe

**Comment by joewhitsitt** [comment]
I've previously messed around with autofilling fields based on dropdown selection: https://github.com/uiowa/uiowa/issues/2504, not sure if i ever pushed a branch though. I will see if I find anything.

Also,
>FWIW, I started a DM with Chris about how the management of content hub’s places content is handled.

**Comment by joewhitsitt** [comment]
Chris responded:
>I am the person who manages that column [content hub places], along with other elements of the taxonomy. Place requests are pretty few and far between; I can't remember the last one that wasn't affiliated with a university location.

**Comment by briand44** [comment]
blocked by determining the appropriate solution that meets the needs of our partners. https://github.com/uiowa/uiowa/issues/6494

**Comment by joewhitsitt** [comment]
I believe #6553 addresses:

> Add replicate functionality to SiteNow events (local drupal events)
> Update company name label to location name
> Add lat/long geofields

So that would just leave:

>Add additional directions field
>Create an internally focused display that shows existing events on their site with the location where they can easily find and get to an appropriate event to replicate

**Comment by briand44** [comment]
grooming conversation... 

possibly use content overview duplicate as a potential path forward. Something similar was done on Brand site for lockup generator. https://brand.uiowa.edu/admin/content/lockups

https://admissions.uiowa.edu/areas-of-study/print-links is an example of an internal display for admissions

**Comment by joewhitsitt** [comment]
Sorry, to clarify, is all of this specific to oniowa or can some of it go into the event feature default?

**Comment by briand44** [comment]
The internal display would be specific but the rest could go into the default as far as I'm concerned. The lat/long wouldn't do anything but I don't know that it would hurt anything. Open to other thoughts

**Comment by joewhitsitt** [comment]
>The internal display would be specific but the rest could go into the default as far as I'm concerned. The lat/long wouldn't do anything but I don't know that it would hurt anything. Open to other thoughts

If lat/lon goes into the default, I think it should be displayed on the full node or we are creating another summary field issue.

**Comment by pyrello** [comment]
> If lat/lon goes into the default, I think it should be displayed on the full node or we are creating another summary field issue.

I think lat/long should be default. If you are saying we should not display a map on the teaser, I agree.

As far as I can tell, this all seems like it should be default material. My thought is that it sounds like stuff that would go into Content Hub 2.0, so we should include it in the default solution for events.

**Comment by joewhitsitt** [comment]
Wasn't thinking a map, but should the lat/long be displayed on the sidebar with the address?

**Comment by pyrello** [comment]
> Wasn't thinking a map, but should the lat/long be displayed on the sidebar with the address?

What would be the purpose? For copying/pasting? I think for most people this information is meaningless without a map program to interpret it

**Comment by joewhitsitt** [comment]
At least short term it would be a field in default that does nothing then which I was trying avoid. Then if only oniowa has the exposed endpoint, help text would be needed to explain that is what it is for.

**Comment by joewhitsitt** [comment]
<img width="260" alt="Screenshot 2023-06-19 at 8 53 05 AM" src="https://github.com/uiowa/uiowa/assets/4663676/7cc642aa-729e-476d-81f3-e49afcaf6e44">

Or could switch it to decimal for easier copy/paste to google maps but DMS works as well. https://www.google.com/maps/place/41%C2%B039'35.0%22N+91%C2%B032'07.0%22W

**Comment by pyrello** [comment]
My opinion is let's only show something if we think it is actually a useful/usable feature. Otherwise it's noise.

**Comment by pyrello** [comment]
To be clear: I think this field should be part of the default. There are definitely usable features we could create from a lat/long, but it sounds like we may not want to prioritize them for the current work. So, lets not have a specific output unless there is something we can produce that adds value. We can add something useful in future work.

**Comment by briand44** [comment]
^^ This has been my thoughts as well
